sdks/java/fn-execution/src/main/java/org/apache/beam/sdk/fn/data/FnDataReceiver.java

```diff
@@ -15,13 +15,11 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package org.apache.beam.runners.fnexecution.data;
-
-import java.io.Closeable;
+package org.apache.beam.sdk.fn.data;
 
 /**
  * A receiver of streamed data.
  */
-public interface FnDataReceiver<T> extends Closeable {
+public interface FnDataReceiver<T> extends AutoCloseable {
```

Comment:
Why would we want to make FnDataReceiver not idempotent on `close()`?
Do we expect the caller to be able to do something internally within the FnDataReceiver to fix it and try again?
Is it so we can throw something other then an `IOException`?

Reply:
The latter. I can add documentation that close should be idempotent to the class and the method, if it's worthwhile.

Reply:
Yes, please add the comment and then LGTM. Afterwards you can self merge.